cdc/log.cc

```diff
@@ -292,6 +292,21 @@ future<> cdc::cdc_service::stop() {
 
 cdc::cdc_service::~cdc_service() = default;
 
+namespace {
+const sstring delta_mode_string_off  = "off";
+const sstring delta_mode_string_keys = "keys";
+const sstring delta_mode_string_full = "full";
+
+sstring to_string(cdc::delta_mode dm) {
+    switch (dm) {
+        case cdc::delta_mode::off  : return delta_mode_string_off;
+        case cdc::delta_mode::keys : return delta_mode_string_keys;
+        case cdc::delta_mode::full : return delta_mode_string_full;
+        default : throw std::runtime_error("Impossible value of cdc::delta_mode");
```

Comment:
No need for default. Compiler should complain about switch not covering all possible enum values.

Reply:
Maybe my compiler is stupid. Without `default`:
```
distcc[638630] ERROR: compile cdc/log.cc on localhost failed
cdc/log.cc: In function ‘seastar::sstring {anonymous}::to_string(cdc::delta_mode)’:
cdc/log.cc:307:1: error: control reaches end of non-void function [-Werror=return-type]
```
I can move `throw` outside of `switch` - just checked that my gcc allows this.

Reply:
Yes. You have to throw unconditionally at the end.